aws/resource_aws_sagemaker_training_job.go

```diff
@@ -0,0 +1,681 @@
+				channel.CompressionType = aws.String(v.(string))
+			}
+			if v, ok := idcItemProperties["record_wrapper_type"]; ok {
+				channel.RecordWrapperType = aws.String(v.(string))
+			}
+			inputDataConfig[idx] = channel
+		}
+	}
+	createOpts.InputDataConfig = inputDataConfig
+
+	var outputDataConfig *sagemaker.OutputDataConfig = new(sagemaker.OutputDataConfig)
+	if odc, ok := d.GetOk("output_data_config"); ok {
+		odcPorperties := odc.([]interface{})
+		properties := odcPorperties[0].(map[string]interface{})
+
+		if v, ok := properties["s3_output_path"]; ok {
+			outputDataConfig.S3OutputPath = aws.String(v.(string))
+		}
+		if v, ok := properties["kms_key_id"]; ok {
+			outputDataConfig.KmsKeyId = aws.String(v.(string))
+		}
+	}
+	createOpts.OutputDataConfig = outputDataConfig
+
+	var resourceConfig *sagemaker.ResourceConfig = new(sagemaker.ResourceConfig)
+	if rc, ok := d.GetOk("resource_config"); ok {
+		rcPorperties := rc.([]interface{})
+		properties := rcPorperties[0].(map[string]interface{})
+
+		if v, ok := properties["instance_type"]; ok {
+			resourceConfig.InstanceType = aws.String(v.(string))
+		}
+		if v, ok := properties["instance_count"]; ok {
+			resourceConfig.InstanceCount = aws.Int64(int64(v.(int)))
+		}
+		if v, ok := properties["volume_size_in_gb"]; ok {
+			resourceConfig.VolumeSizeInGB = aws.Int64(int64(v.(int)))
+		}
+	}
+	createOpts.ResourceConfig = resourceConfig
+
+	var stoppingCondition *sagemaker.StoppingCondition = new(sagemaker.StoppingCondition)
+	if sc, ok := d.GetOk("stopping_condition"); ok {
+		scPorperties := sc.([]interface{})
+		properties := scPorperties[0].(map[string]interface{})
+
+		if v, ok := properties["max_runtime_in_seconds"]; ok {
+			stoppingCondition.MaxRuntimeInSeconds = aws.Int64(int64(v.(int)))
+		}
+	}
+	createOpts.StoppingCondition = stoppingCondition
+
+	if v, ok := d.GetOk("tags"); ok {
+		tagsIn := v.(map[string]interface{})
+		createOpts.Tags = tagsFromMapSagemaker(tagsIn)
+	}
+
+	log.Printf("[DEBUG] Sagemaker Training Job create config: %#v", *createOpts)
+	createResponse, err := conn.CreateTrainingJob(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating Sagemaker Training Job: %s", err)
+	}
+
+	d.SetId(name)
+	if err := d.Set("arn", createResponse.TrainingJobArn); err != nil {
+		return err
+	}
+	log.Printf("[INFO] Sagemaker Training Job ID: %s", d.Id())
+
+	return resourceAwsSagemakerTrainingJobRead(d, meta)
+}
+
+func resourceAwsSagemakerTrainingJobRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	trainingJobRaw, _, err := SagemakerTrainingJobStateRefreshFunc(conn, d.Id())()
+	if err != nil {
+		return err
+	}
+	if trainingJobRaw == nil {
+		d.SetId("")
+		return nil
+	}
+
+	trainingJob := trainingJobRaw.(*sagemaker.DescribeTrainingJobOutput)
+
+	if err := d.Set("name", trainingJob.TrainingJobName); err != nil {
+		log.Printf("[ERR] Error setting Name: %s", err)
```

Comment:
As noted in the instance resource, we should combine the logs and error into the return